devlib/utils/android.py

```diff
@@ -312,18 +312,19 @@ def _ping(device):
 
 
 def adb_shell(device, command, timeout=None, check_exit_code=False,
-              as_root=False, newline_separator='\r\n'):  # NOQA
+              as_root=False, newline_separator='\r\n', exec_func=check_output):  # NOQA
     _check_env()
     if as_root:
+        orig_command = 'echo \'{}\' | su'.format(command)
         command = 'echo \'{}\' | su'.format(escape_single_quotes(command))
     device_string = ' -s {}'.format(device) if device else ''
     full_command = 'adb{} shell "{}"'.format(device_string,
                                              escape_double_quotes(command))
     logger.debug(full_command)
     if check_exit_code:
         actual_command = "adb{} shell '({}); echo \"\n$?\"'".format(device_string,
-                                                                    escape_single_quotes(command))
-        raw_output, error = check_output(actual_command, timeout, shell=True)
+                                                                    escape_single_quotes(orig_command if as_root else command))
```

Comment:
Rather than saving `orig_command`, why not  just do

      command if as_root else escape_single_quotes(command))

here?